tests/integration/test_s3.py

```diff
@@ -117,8 +140,8 @@ def test_s3_put_object_notification(self):
         obj = self.s3_client.put_object(Bucket=bucket_name, Key=key_by_path, Body='something')
 
         # put an object where the bucket_name is in the host
-        # it doesn't care about the authorization header as long as it's present
-        headers = {'Host': '{}.s3.amazonaws.com'.format(bucket_name), 'authorization': 'some_token'}
+        headers = aws_stack.mock_aws_request_headers('s3')
+        headers['Host'] = '%s.s3.localhost.localstack.cloud:4566' % bucket_name
```

Comment:
Same as above - Let's better use `S3_VIRTUAL_HOSTNAME` and `EDGE_PORT` here, to avoid hardcoding values.